### PR TITLE
Update for cmdlfnexwatch.c

### PR DESCRIPTION
**Pull request by Guilhem7**
Hi Iceman, this modification allows a user to find the magic number and to use the psk2 modulation to clone a t55 tag. Moreover the magic byte can also be added in parameter of the clone command if it isn't a known credential.

### CODE REVIEW

File: common/lfdemod.c [hunk 1]
```diff
@@ -295,6 +295,17 @@ int bits_to_array(const uint8_t *bits, size_t size, uint8_t *dest) {
     return PM3_SUCCESS;
 }
 
+// Ex: 0x56 and numbytes = 1 ==> return res = {0,1,0,1,0,1,1,0}
```
Comment (iceman1001) [inline common/lfdemod.c:298]: There is already a  `bytes_to_bytebits` fct in util.c    Use it instead

File: client/src/cmdlfnexwatch.c [hunk 2]
```diff
@@ -108,6 +109,21 @@ static int nexwatch_scamble(NexWatchScramble_t action, uint32_t *id, uint32_t *s
     return PM3_SUCCESS;
 }
 
+static int nexwatch_magic_bruteforce(uint32_t cn, uint8_t calc_parity, uint8_t chk) {
+    uint8_t magic = 0;
+    uint8_t temp_checksum;
+    for (; magic < 255; magic++) {
```
Comment (iceman1001) [inline client/src/cmdlfnexwatch.c:115]: cppchecker will say that you can lessen the scope of these two vars.   ie,  move them inside.

File: client/src/cmdlfnexwatch.c [hunk 3]
```diff
@@ -285,6 +304,8 @@ static int CmdNexWatchClone(const char *Cmd) {
         arg_lit0(NULL, "hc", "Honeywell credential"),
         arg_lit0(NULL, "q5", "optional - specify writing to Q5/T5555 tag"),
         arg_lit0(NULL, "em", "optional - specify writing to EM4305/4469 tag"),
+        arg_str0(NULL, "magic", "<hex>", "optional - magic hex data. 1 byte"),
```
Comment (iceman1001) [inline client/src/cmdlfnexwatch.c:307]: add `m"  - for a short param

File: client/src/cmdlfnexwatch.c [hunk 4]
```diff
@@ -368,6 +407,17 @@ static int CmdNexWatchClone(const char *Cmd) {
         raw[10] |= nexwatch_checksum(magic, cn, parity);
     }
 
+    if (use_psk2) {
+        blocks[0] = 270464;
+        uint8_t * res_shifted = malloc(96 * sizeof(uint8_t));
```
Comment (iceman1001) [inline client/src/cmdlfnexwatch.c:412]: 1. we use calloc in the repo instead of malloc.
2. I see no free of this allocated memory

**Review by iceman1001 — CHANGES_REQUESTED** [review]

**Comment by iceman1001** [comment]
Awesome!  
Some minor stuff to fix and its mergable

File: client/src/cmdlfnexwatch.c (hunk 3 above)
Comment (Guilhem7) [reply client/src/cmdlfnexwatch.c:307]: but m is already used for mode.

File: client/src/cmdlfnexwatch.c (hunk 4 above)
Comment (Guilhem7) [reply client/src/cmdlfnexwatch.c:412]: Ok, we fix it this afternoon then !

File: client/src/cmdlfnexwatch.c (hunk 3 above)
Comment (iceman1001) [reply client/src/cmdlfnexwatch.c:307]: I see.  bugger.    Ignore this one :)

File: client/src/cmdlfnexwatch.c (hunk 2 above)
Comment (Guilhem7) [reply client/src/cmdlfnexwatch.c:115]: do you mean we can remove the "static" ?

**Pull request closed without merging**